zachsinterview/src/main/java/com/imprev/zachsinterview/Main.java

```diff
@@ -0,0 +1,37 @@
+package com.imprev.zachsinterview;
+
+public class Main {
+
+    /**
+     * Demonstrates how to connect to thirdPartyAPI and tests how the numerical key store may be used.
+     */
+    public static void main(String[] argv) throws NumericalKeyValueException {
+        
+        String username = "moxiworks";
+        String password = "fjdskal3kd!jdkdInterviewFun!!!";
+        String apiUrl = "https://example.com/thirdPartyApiURL";
+        
+        NumericalKeyValueStore keyStore = new NumericalKeyValueStoreImpl(username, password, apiUrl , 80);
+        NumericalKeyValueFun keyFun = new NumericalKeyValueFunImpl(keyStore);
+        
```

Comment:
Why don't we add a test for valid credentials here so that we bail out at the beginning.